include/constants/savedata/savedata_misc.h

```diff
@@ -0,0 +1,7 @@
+#ifndef POKEPLATINUM_CONSTANTS_SAVEDATA_MISC_H
+#define POKEPLATINUM_CONSTANTS_SAVEDATA_MISC_H
+
+#define RIVAL_NAME_LEN      7
+#define TABLET_NAME_LEN     10
```

Comment:
There is a header `constants/string.h` that is meant to be used to store string-length constants like these.

I think `RIVAL_NAME_LEN` is thus superfluous, as it's the same length as any other trainer.